### ISSUE TITLE
Accurate Slapshot: Level select screen does not load

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#8766]
### Describe the bug

Link to SWF: http://i.notdoppler.com/files/accurateslapshot.swf
Note: Requires a stub of `SharedObject.size`.

In [Accurate Slapshot](http://www.notdoppler.com/accurateslapshot.php), there are 2 related issues:
- Clicking the Continue button on the main menu goes to a black screen.
- Clicking the X button in the top-right corner during gameplay softlocks the game.
![image](https://user-images.githubusercontent.com/71368227/207266942-9edf78e7-26c2-4bed-992e-04888df4f6e7.png)

Both buttons are supposed to go to the level select screen. And in both cases the following error is shown in the terminal:
`[ERROR ruffle_core::avm2::events] Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x29adda9bf20 }) to handler FunctionObject(FunctionObject(GcCell(Gc { ptr: 0x29ac48bac88 }))) : RustError("Cannot access property ::length of null or undefined")`

### Expected behavior

Both buttons should take you to the level select screen.

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

Both problems are caused by code in the game's `TweenMax` package. Here is an SWF with many trace statements added to demonstrate: [accurateslapshot_trace.zip](https://github.com/ruffle-rs/ruffle/files/10216185/accurateslapshot_trace.zip)

After clicking the Continue button, the `showLevelManager()` function runs:
```
         if(contains(this._menu))
         {
            trace("About to run killChildTweensOf");
            TweenMax.killChildTweensOf(this._menu,true);
            trace("Just ran killChildTweensOf");
            removeChild(this._menu);
         }
```
Notice that in Ruffle, the first trace is shown right before the error, and the second trace never shows.

The `killChildTweensOf` function calls `getAllTweens()`:
```
         trace("About to run getAllTweens");
         var _loc3_:Array = getAllTweens();
         trace("Just ran getAllTweens");
```
Notice that in Ruffle, the first trace is shown right before the error, and the second trace never shows.

And finally in `getAllTweens`, we have the offending use of `length`:
```
         var _loc4_:Array = null;
         var _loc5_:int = 0;
         var _loc1_:Dictionary = masterList;
         var _loc2_:int = 0;
         var _loc3_:Array = [];
         for each(_loc4_ in _loc1_)
         {
            _loc5_ = int(_loc4_.length);
            while(--_loc5_ > -1)
            {
               if(!TweenLite(_loc4_[_loc5_]).gc)
               {
                  var _loc8_:*;
                  _loc3_[_loc8_ = _loc2_++] = _loc4_[_loc5_];
               }
            }
         }
```
This code is very difficult to read, and I can't figure out why `_loc4_.length` causes the error, but it does.

**Comment by n0samu** [comment]
The same error is displayed when trying to start the game in [Koutack](http://www.notdoppler.com/koutack.php) and when trying to start a level in [Track the Ball](http://www.notdoppler.com/tracktheball.php). Both games use tweening when transitioning between screens. Is it the same problem?

**Comment by Benman2785** [comment]
same problem with Road_of_the_Dead.swf

screen goes white 
[link to swf](https://hoffmeister.li/flash-games/swf/Road_of_the_Dead.swf)

**Comment by n0samu** [comment]
Update: I found that the issue with Koutack is indeed the same problem. On the other hand, the issue with Track the Ball occurs because the levels are stored as XML data, and the required XML functions are not yet implemented.

**Comment by n0samu** [comment]
> same problem with Road_of_the_Dead.swf
> 
> screen goes white [link to swf](https://hoffmeister.li/flash-games/swf/Road_of_the_Dead.swf)

This does indeed seem to be the same problem - it's also fixed by #9007!
